gymnasium_robotics/envs/maze/ant_maze.py

```diff
@@ -271,21 +272,27 @@ def reset(self, *, seed: Optional[int] = None, **kwargs):
 
         obs, info = self.ant_env.reset(seed=seed)
         obs_dict = self._get_obs(obs)
+        info["success"] = bool(
+            np.linalg.norm(obs_dict["achieved_goal"] - self.goal) <= 0.45
+        )
 
         return obs_dict, info
 
     def step(self, action):
         ant_obs, _, _, _, info = self.ant_env.step(action)
         obs = self._get_obs(ant_obs)
 
+        info["success"] = bool(np.linalg.norm(obs["achieved_goal"] - self.goal) <= 0.45)
+        reward = self.compute_reward(obs["achieved_goal"], self.goal, info)
         terminated = self.compute_terminated(obs["achieved_goal"], self.goal, info)
         truncated = self.compute_truncated(obs["achieved_goal"], self.goal, info)
 
-        reward = self.compute_reward(obs["achieved_goal"], self.goal, info)
-
         if self.render_mode == "human":
             self.render()
 
+        # Update the goal position if necessary
+        self.update_goal(obs["achieved_goal"], self.goal)
```

Comment:
Should be after `truncated `, before `info`

Reply:
I would think that moving `update_goal()` before `info` would mean that `info['success'] = false` always?

Reply:
@alexdavey is right here, we should keep it as it is